src/App.jsx

```diff
@@ -1,17 +1,40 @@
 import './App.css';
+import messages from './data/messages.json';
+import ChatLog from './components/ChatLog';
+import { useState } from 'react';
+
+
+function App() {
+  const [chatData, setChatData] = useState(messages);
+  const [likesCount, setLikesCount] = useState(0);
+
+  const toggleLiked = (chatEntryId) => {
+    setChatData(chats => {
+      return chats.map(chat => {
+        if (chat.id === chatEntryId) {
+          return { ...chat, liked: !chat.liked };
+        } else {
+          return chat;
+        }
+      });
+    });
+  };
+
+  const calculateLikes = (isLiked) => {
+    setLikesCount((likesCount) => isLiked ? likesCount - 1 : likesCount + 1);
+  };
 
-const App = () => {
   return (
     <div id="App">
       <header>
-        <h1>Application title</h1>
+        <h1>Chat Between Vladinir and Estragon</h1>
+        <p>{likesCount} ❤️s</p>
       </header>
       <main>
-        {/* Wave 01: Render one ChatEntry component
-        Wave 02: Render ChatLog component */}
+        <ChatLog entries={chatData} onLikedToggle={toggleLiked} likesCount={likesCount} setLikesCount={setLikesCount} calculateLikes={calculateLikes}/>
```

Comment:
Nothing wrong with what you did functionally, this is more so for readability purposes.
```suggestion
        <ChatLog 
        entries={chatData} 
        onLikedToggle={toggleLiked} 
        likesCount={likesCount} 
        setLikesCount={setLikesCount} 
        calculateLikes={calculateLikes}/>
```